### DISCUSSION TITLE
Support the basic markdown task types

### DISCUSSION THREAD

**Discussion by benhamilton** [roovo/obsidian-card-board#66] [Ideas]
Per https://github.com/obsidian-community/theme-dev-vault/blob/main/List%20Variations.md#task-lists

include the following task types as incomplete, and therefore displayed in the columns:
- [ ] incomplete
- [>] > deferred
- [/] / for in progress, or half-done
- [!] ! for Important
- [?] ? for question

but possibly not display:
- [-] - cancelled

and only display these in the done/completed column:
- [x] x complete / done
- [X] X complete / done (Failed in Sanctum)

**Comment by roovo** [comment]
I have got a note to look at alternative task types and didn't realise that there are some basic ones defined - that's helpful.  Something along these lines would certainly be possible to do.

I am wondering (out loud with no thought really) whether it could be sensible or not to treat anything that is a single character contained in a "- [ ] " as the first non-whitespace stuff on a line as a task?  Then count everything as incomplete apart from those with x's or X's in them (and possibly cancelled) as incomplete.  That way, whatever system people want to use or whatever themes want to do it would be supported out the box.

What danger would there be of me capturing non-tasks as tasks?  (I'm asking myself as well)

**Reply by benhamilton** [reply]
1. Supporting alternate task types by treating incomplete tasks as any task without a `x` or `X` is clear... except for tasks that are cancelled, which does seem to be treated universally as `- [-] ` - I can't find an exception to that (yet).
2. I'm 100% in favour of capturing non-markdown tasks as tasks. See my [comment here](https://github.com/roovo/obsidian-card-board/discussions/65#discussioncomment-1802740) for more detail as to why. In an _agile_ or _scrum_ environment, cards on a board don't have to be a _task_, they could be a _epic, _story_, _task_, or _sub-task_. Markdown tasks really only account for the _task_ and the _sub-task_ (via indentation).
